test/extensions/filters/network/common/fuzz/uber_filter.cc

```diff
@@ -0,0 +1,108 @@
+  ON_CALL(read_filter_callbacks_->connection_, requestedServerName())
+      .WillByDefault(testing::Return("fake_cluster"));
+  // Prepare time source for filters such as local_ratelimit filter.
+  factory_context_.prepareSimulatedSystemTime();
+  // Prepare address for filters such as ext_authz filter.
+  addr_ = std::make_shared<Network::Address::PipeInstance>("/test/test.sock");
+  read_filter_callbacks_->connection_.remote_address_ = addr_;
+  read_filter_callbacks_->connection_.local_address_ = addr_;
+  async_request_ = std::make_unique<Grpc::MockAsyncRequest>();
+}
+
+UberFilterFuzzer::UberFilterFuzzer() : time_source_(factory_context_.simulatedTimeSystem()) {
+  fuzzerSetup();
+}
+
+void UberFilterFuzzer::fuzz(
+    const envoy::config::listener::v3::Filter& proto_config,
+    const Protobuf::RepeatedPtrField<::test::extensions::filters::network::Action>& actions) {
+  try {
+    // Try to create the filter callback(cb_). Exit early if the config is invalid or violates PGV
+    // constraints.
+    const std::string& filter_name = proto_config.name();
+    ENVOY_LOG_MISC(info, "filter name {}", filter_name);
+    auto& factory = Config::Utility::getAndCheckFactoryByName<
+        Server::Configuration::NamedNetworkFilterConfigFactory>(filter_name);
+    ProtobufTypes::MessagePtr message = Config::Utility::translateToFactoryConfig(
+        proto_config, factory_context_.messageValidationVisitor(), factory);
+    // Make sure no invalid system calls are executed in fuzzer.
+    checkInvalidInputForFuzzer(filter_name, message.get());
+    ENVOY_LOG_MISC(info, "Config content after decoded: {}", message->DebugString());
+    cb_ = factory.createFilterFactoryFromProto(*message, factory_context_);
+
+  } catch (const EnvoyException& e) {
+    ENVOY_LOG_MISC(debug, "Controlled exception in filter setup{}", e.what());
+    return;
+  }
+  perFilterSetup(proto_config.name());
+  // Add filter to connection_.
+  cb_(read_filter_callbacks_->connection_);
+  for (const auto& action : actions) {
+    ENVOY_LOG_MISC(trace, "action {}", action.DebugString());
+    switch (action.action_selector_case()) {
+    case test::extensions::filters::network::Action::kOnData: {
+      ASSERT(read_filter_ != nullptr);
+      Buffer::OwnedImpl buffer(action.on_data().data());
+      read_filter_->onData(buffer, action.on_data().end_stream());
+
+      break;
+    }
+    case test::extensions::filters::network::Action::kOnNewConnection: {
+      ASSERT(read_filter_ != nullptr);
+      read_filter_->onNewConnection();
+
+      break;
+    }
+    case test::extensions::filters::network::Action::kAdvanceTime: {
+      time_source_.advanceTimeAsync(
+          std::chrono::milliseconds(action.advance_time().milliseconds()));
+      factory_context_.dispatcher().run(Event::Dispatcher::RunType::NonBlock);
+      break;
+    }
+    default:
+      // Unhandled actions.
+      PANIC("A case is missing for an action");
```

Comment:
Say which action is missing.  action.DebugString() gives you a usable string.

Reply:
Thanks! Added!